senders/msteams/msteams.go

```diff
@@ -0,0 +1,213 @@
+package msteams
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/moira-alert/moira"
+	"github.com/russross/blackfriday/v2"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Sender implements moira sender interface via MS Teams
+type Sender struct {
+	frontURI string
+	logger   moira.Logger
+	location *time.Location
+	client   *http.Client
+}
+
+// Init initialises settings required for full functionality
+func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+	sender.logger = logger
+	sender.location = location
+	sender.frontURI = senderSettings["front_uri"]
+	disableKeepAlives, err := strconv.ParseBool(senderSettings["disable_keepalives"])
+	if err != nil {
+		disableKeepAlives = false
+	}
+	sender.client = &http.Client{
+		Timeout:   time.Duration(30) * time.Second,
+		Transport: &http.Transport{DisableKeepAlives: disableKeepAlives},
```

Comment:
I still don't see the cases where we need this functionality/ But maybe I am wrong. @beevee what do you think about it?

Reply:
@imavroukakis can you please explain, why would anybody want to disable keepalives? Is there a reason (maybe a bug in MS Teams API or something)?

Reply:
No bug that I'm aware of @beevee - I don't have a strong argument for or against this, except this: You _could_ argue, that with keep-alive turned on, you could eventually reach a connection limit on the MS server. Since this is an unknown value, it should at least be made configurable for users with a lot of webhooks. I think, enabling keep-alives by default is a sane strategy, with the option to turn them off if connection capacity issues are encountered.

Reply:
As far as I understand, enabling keepalives will result in one (?) established connection per one notifier instance. For all practical purposes, there is no need to keep more than 2 or 3 notifier instances running. We only have one, since our Kubernetes cluster handles automatic failover, and high load is not an issue here.

If you ask me, I'd say that adding clutter to configuration files and code is not worth having even 3 long-running connections to MS Teams server.

Reply:
I'm happy to take it out then :)

Reply:
Done with https://github.com/moira-alert/moira/pull/432/commits/6f4045397f0f583d97fdadf299518e8ac8177ea4